test/powershell/Language/Scripting/CmdletDeclaration.Test.ps1

```diff
@@ -0,0 +1,215 @@
+                    param([Parameter(mandatory=$true)]$a)
+                    $a
+                }'
+
+            $null = [system.management.automation.psparser]::tokenize($script, [ref] $syntaxerrors)
+
+            #Error should not be reported
+            $syntaxerrors.Count | Should Be 0
+        }
+
+        It 'Cmdlet declaration parameter: defaultparametersetname' {
+
+            $syntaxerrors = $null
+            $script = '
+                function get-foo
+                {
+                    [CmdletBinding(SupportsShouldProcess=$true, ConfirmImpact="low", defaultparametersetname="set1")]
+                    param([Parameter(mandatory=$true)]$a)
+                    $a
+                }'
+
+            $null = [system.management.automation.psparser]::tokenize($script, [ref] $syntaxerrors)
+
+            #Error should not be reported
+            $syntaxerrors.Count | Should Be 0
+        }
+
+        It 'cmdlet declaration within as cmdlet' {
+
+            $syntaxerrors = $null
+
+            $script = '
+                function get-foo {
+                    [CmdletBinding()]
+                    param()
+
+	            function get-bar
+                    {
+                        param([Parameter(mandatory=$true)]$a)
+                        $a
+                    }
+                }'
+
+            $null = [system.management.automation.psparser]::tokenize($script, [ref] $syntaxerrors)
+
+            #Error should not be reported
+            $syntaxerrors.Count | Should Be 0
+        }
+
+        It 'cmdlet declaration within scripblock' {
+
+            $syntaxerrors = $null
+            $script = '
+                {
+                    param([Parameter(mandatory=$true)]$a)
+                    $a
+                }'
+
+            $null = [system.management.automation.psparser]::tokenize($script, [ref] $syntaxerrors)
+
+            #Error should not be reported
+            $syntaxerrors.Count | Should Be 0 
+        }
+
+        It 'cmdlet declaration within scripblock 2' {
+
+            $syntaxerrors = $null
+
+            $script = '
+                {
+                    param([Parameter(mandatory=$true)]$a)
+                    $a
+                }'
+
+            $null = [system.management.automation.psparser]::tokenize($script, [ref] $syntaxerrors)
+
+            #Error should not be reported
+            $syntaxerrors.Count | Should Be 0
+        }
+
+        It 'cmdlet declaration within scripblock 3' {
+        
+            $syntaxerrors = $null
+
+            $script = '
+                {
+                    [CmdletBinding(SupportsShouldProcess=$true)]
+                    param([Parameter(mandatory=$true)]$a)
+                    $a
+                }'
+
+            $null = [system.management.automation.psparser]::tokenize($script, [ref] $syntaxerrors)
+
+            #Error should not be reported
+            $syntaxerrors.Count | Should Be 0
+        }
```

Comment:
it seems like all of these variations should be done as testcases, rather than individual tests.